File: Revisao/incremental7/CartaoEmpresarial.java
class CartaoEmpresarial extends CartaoDeCredito {
    private float limiteExtra;

    public CartaoEmpresarial(String nomeTitular, String numero, float limite, float limiteExtra, Fatura fatura) {
        super(nomeTitular, numero, limite, fatura);
        this.limiteExtra = limiteExtra;
    }

    @Override
    public void realizarCompra(float valor) {
        if (!ativo) {
            System.out.println("Cartão inativo. Transação não permitida.");
            return;
        }

        float saldoDisponivel = limite + limiteExtra;

        if (saldo + valor <= saldoDisponivel) {
            saldo += valor;
            Transacao transacao = new Transacao(new Date(), valor, "Compra empresarial", "compra");
            adicionarTransacao(transacao);
            System.out.println("Compra de R$" + valor + " realizada com sucesso no Cartão Empresarial.");
        } else {
            System.out.println("Compra não autorizada. Limite insuficiente.");
        }
    }

    public void realizarCompraParcelada(float valor, int parcelas) {
        if (!ativo) {
            System.out.println("Cartão inativo. Transação não permitida.");
            return;
        }

        float valorParcela = valor / parcelas;
        float saldoDisponivel = limite + limiteExtra;

        if (saldo + valor <= saldoDisponivel) {
            saldo += valor;
            for (int i = 1; i <= parcelas; i++) {
                Transacao transacao = new Transacao(new Date(), valorParcela, "Parcela " + i + "/" + parcelas, "compra parcelada");
                adicionarTransacao(transacao);
            }
            System.out.println("Compra parcelada de R$" + valor + " em " + parcelas + "x de R$" + valorParcela + " realizada com sucesso.");
        } else {
            System.out.println("Compra não autorizada. Limite insuficiente.");
        }
    }
}

File: Revisao/python/cartaoEmpresarial.py

import datetime
from Revisao.python import transacao
from Revisao.python.cartaoDeCredito import CartaoDeCredito
from Revisao.python.transacao import Transacao


class CartaoEmpresarial(CartaoDeCredito):
    def __init__(self, nome_titular, numero, limite, fatura, limite_extra):
        super().__init__(nome_titular, numero, limite, fatura)
        self.limite_extra = limite_extra
    
    def realizar_compra(self, valor):
        if not self.ativo:
            print("Cartão inativo")
            return
        saldo_disponivel = self.limite + self.limite_extra
        if self.saldo + valor <= saldo_disponivel:
            self.saldo += valor
            trans = Transacao(datetime.now(), valor, "compra", "debito")
            self.adicionar_transacao(transacao)
            print("Compra de R${valor} realizada com sucesso")
        else:
            print("Saldo insuficiente")
            
    def realizar_compra_parcelada(self, valor, parcelas):
        if not self.ativo:
            print("Cartão inativo")
            return
        valor_parcela = valor / parcelas
        saldo_disponivel = self.limite + self.limite_extra
        if self.saldo + valor <= saldo_disponivel:
            self.saldo += valor
            for i in range(1, parcelas + 1):
                trans = Transacao(datetime.now(), valor_parcela, f"Parcela {i}/{parcelas}", "compra parcelada")
            print(f"Compra de R${valor} realizada em {parcelas} parcelas de R${valor_parcela} cada")
        else:
            print("Limite insuficiente")
            